### PR TITLE
[dev-v5] Add FluentDivider component

### PR DESCRIPTION
**Pull request by AClerbois**
# Add FluentDivider component

The **FluentDivider** component is used to separate content within a container.

## Horizontally
{1651DE57-82DB-4F26-AB65-AEEB9FF87992}

## Vertically

{A0190758-CCEB-456F-9356-B95088199646}

## Documentation

- **AlignContent** Gets or sets the alignment of the content.
- **Appearance** Gets or sets the appearance of the content.
- **ChildContent** Gets or sets the content to be shown.
- **Inset** Adds padding to the beginning and end of the divider.
- **Vertical** A divider can be horizontal (default) or vertical.

## Unit Tests

{415DBAF9-31E5-4F6F-A47D-C08F49C9D169}

### CODE REVIEW

File: src/Core/Enums/ButtonAppearance.cs [hunk 1]
```diff
@@ -34,4 +34,3 @@ public enum ButtonAppearance
     /// </summary>
     Transparent,
 }
```
Comment (dvoituron) [inline src/Core/Enums/ButtonAppearance.cs:36]: Why did you update this file ?

File: src/Core/Enums/DividerAppearance.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public enum DividerAppearance
+{
+    /// <summary>
+    /// The divider appears with the default style.
+    /// </summary>
+    [Description("")]
+    Default,
+
+    /// <summary>
+    /// The divider appears with the default style.
```
Comment (dvoituron) [inline src/Core/Enums/DividerAppearance.cs:21]: Probably "strong" style?

File: tests/Core/Components/Divider/FluentDividerTests.razor [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        var cut = Render(@<FluentDivider>Hello World</FluentDivider>);
+
+        // Assert
+        cut.Verify();
+    }
+
+    [Theory]
+    [InlineData(DividerAlignContent.Center, "center")]
+    [InlineData(DividerAlignContent.Start, "start")]
+    [InlineData(DividerAlignContent.End, "end")]
```
Comment (dvoituron) [inline tests/Core/Components/Divider/FluentDividerTests.razor:27]: You need to also test:
 - `[InlineData(null, "center")]`
 - `[InlineData((DividerAlignContent)999, "center")]`

File: tests/Core/Components/Divider/FluentDividerTests.razor [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        // Assert
+        var attribute = cut.Find("fluent-divider").GetAttribute("align-content") ?? string.Empty;
+        Assert.Equal(expectedValue, attribute);
+    }
+
+    [Theory]
+    [InlineData(DividerAppearance.Default, "")]
+    [InlineData(DividerAppearance.Subtle, "subtle")]
+    [InlineData(DividerAppearance.Brand, "brand")]
+    [InlineData(DividerAppearance.Strong, "strong")]
```
Comment (dvoituron) [inline tests/Core/Components/Divider/FluentDividerTests.razor:42]: Same, you need to test `null` and `(DividerAppearance)999` which are possible values

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/Core/Enums/ButtonAppearance.cs (hunk 1 above)
Comment (AClerbois) [reply src/Core/Enums/ButtonAppearance.cs:36]: I didn't, may be a refactor on save happens on this file

File: src/Core/Enums/DividerAppearance.cs (hunk 2 above)
Comment (AClerbois) [reply src/Core/Enums/DividerAppearance.cs:21]: Well seen

**Review by dvoituron — APPROVED** [review]

**Review by vnbaaij — APPROVED** [review]
